core/src/main/java/tachyon/worker/BlockHandler.java

```diff
@@ -18,86 +18,82 @@
 import java.io.Closeable;
 import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.nio.channels.ByteChannel;
+
+import io.netty.channel.FileRegion;
 
 import tachyon.TachyonURI;
 
 /**
- * Base class for handling block I/O. Block handlers for different under file systems can be
- * implemented by extending this class. It is not thread safe, the caller must guarantee thread
- * safe. This class is internal and subject to changes.
+ * General interface for handling block I/O. Block handlers for different under file systems
+ * implement this interface. It is not thread safe, the caller must guarantee thread safe.
+ * This interface is internal and subject to changes.
  */
-public abstract class BlockHandler implements Closeable {
+public interface BlockHandler extends Closeable {
 
-  /**
-   * Create a block handler according to path scheme
-   * 
-   * @param path the block path
-   * @return the handler of the block
-   * @throws IOException
-   * @throws IllegalArgumentException
-   */
-  public static BlockHandler get(String path) throws IOException, IllegalArgumentException {
-    if (path.startsWith(TachyonURI.SEPARATOR) || path.startsWith("file://")) {
-      return new BlockHandlerLocal(path);
+  class Factory {
```

Comment:
Oh okay, I was just wondering if you had any special reason. We can leave it as is and refactor all of these patterns later if we decide to separate them.